### PR TITLE
Refactor classes

### PR DESCRIPTION
**Pull request by madsonic**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,53 +1,47 @@
...
 
-// this section of code initializes Accuser for the current semester's repository
-// currentLevel in data.json determines which highest level repository is already
-//    made available to the students.
-// previous semesters are no longer handled by this bot.
-let initializeSemesterRepositories = require('./src/semester');
-for (var level = 1; level <= currentLevel; ++level) {
-  initializeSemesterRepositories(accuser, 'addressbook-level' + level);
+// Whitelisted
+const SubmissionRepos = require('./src/whitelist');
```
Comment (louislai) [inline index.js]: hmm i think convention maybe to name the file the same as you name your class, so if your class is `WhiteList` the file will be `WhiteList.js`

Comment (louislai) [reply index.js]: Btw better to move all the require to the top of the file

File: index.js [hunk 2]
```diff
@@ -1,53 +1,47 @@
...
-// upstream se-edu and semester repo
-let initializePracticeForkRepository = require('./src/practiceFork');
-let practiceRepositories = ['samplerepo-pr-practice'];
-let practiceAccounts = ['se-edu', semesterAccount];
-practiceAccounts.forEach(account => {
-  practiceRepositories.forEach(repo => initializePracticeForkRepository(accuser, account, repo));
+// Blacklisted
+const BlackListed = require('./src/blacklist');
+const blackListedRepos = [
+  'sameplrepo-pr-practice',
```
Comment (louislai) [inline index.js]: i guess we do the non-hardcode way later? Also typo here, `samplerepo` not `samplrepo`

File: test/titleChecker.test.js [hunk 3]
```diff
@@ -1,40 +0,0 @@
-const Checker = require('../src/TitleChecker');
```
Comment (louislai) [inline test/titleChecker.test.js]: add back some other tests?

File: src/Validator.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  static checkTitle(title, pattern) {
+    return pattern.exec(title);
+  }
+
+  // check if label is contain in list of labels from issue
+  // issue: labels from this issue
+  // label: label to search for
+  static hasLabel(issue, labelName) {
+    return issue.labels.map(issueLabel => issueLabel.name.toLowerCase())
+      .map(issueLabelName => issueLabelName === labelName.toLowerCase())
```
Comment (louislai) [inline src/Validator.js]: seem like just a `.include(labelName.toLowerCase()` will suffice here?

File: test/titleChecker.test.js (hunk 3 above)
Comment (madsonic) [reply test/titleChecker.test.js]: tests are now `Validator.test.js`

File: index.js [hunk 5]
```diff
@@ -1,53 +1,45 @@
 // Load dotenv first
 require('dotenv').config({ silent: true });
-
-const utility = require('./src/utility');
+const SubmissionRepos = require('./src/Whitelist');
+const BlackListed = require('./src/Blacklist');
```
Comment (louislai) [inline index.js]: wait you changed the names here but the file names themselves are not changed?

Comment (madsonic) [reply index.js]: ahh my git was ignoring case changes

Comment (louislai) [reply index.js]: ah unix system

**Pull request merged**